dotnet/samples/Concepts/ChatCompletion/Connectors_WithMultipleLLMs.cs

```diff
@@ -1,82 +1,156 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
 using Microsoft.SemanticKernel;
-using xRetry;
 
 namespace ChatCompletion;
 
 public class Connectors_WithMultipleLLMs(ITestOutputHelper output) : BaseTest(output)
 {
-    /// <summary>
-    /// Show how to run a prompt function and specify a specific service to use.
-    /// </summary>
-    [RetryFact(typeof(HttpOperationException))]
-    public async Task RunAsync()
+    private const string ChatPrompt = "Hello AI, what can you do for me?";
+
+    private static Kernel BuildKernel()
     {
-        Kernel kernel = Kernel.CreateBuilder()
-            .AddAzureOpenAIChatCompletion(
-                deploymentName: TestConfiguration.AzureOpenAI.ChatDeploymentName,
-                endpoint: TestConfiguration.AzureOpenAI.Endpoint,
-                apiKey: TestConfiguration.AzureOpenAI.ApiKey,
-                serviceId: "AzureOpenAIChat",
-                modelId: TestConfiguration.AzureOpenAI.ChatModelId)
-            .AddOpenAIChatCompletion(
-                modelId: TestConfiguration.OpenAI.ChatModelId,
-                apiKey: TestConfiguration.OpenAI.ApiKey,
-                serviceId: "OpenAIChat")
-            .Build();
-
-        await RunByServiceIdAsync(kernel, "AzureOpenAIChat");
-        await RunByModelIdAsync(kernel, TestConfiguration.OpenAI.ChatModelId);
-        await RunByFirstModelIdAsync(kernel, "gpt-4-1106-preview", TestConfiguration.AzureOpenAI.ChatModelId, TestConfiguration.OpenAI.ChatModelId);
+        return Kernel.CreateBuilder()
+                    .AddAzureOpenAIChatCompletion(
+                        deploymentName: TestConfiguration.AzureOpenAI.ChatDeploymentName,
+                        endpoint: TestConfiguration.AzureOpenAI.Endpoint,
+                        apiKey: TestConfiguration.AzureOpenAI.ApiKey,
+                        serviceId: "AzureOpenAIChat",
+                        modelId: TestConfiguration.AzureOpenAI.ChatModelId)
+                    .AddOpenAIChatCompletion(
+                        modelId: TestConfiguration.OpenAI.ChatModelId,
+                        apiKey: TestConfiguration.OpenAI.ApiKey,
+                        serviceId: "OpenAIChat")
+                    .Build();
     }
 
-    private async Task RunByServiceIdAsync(Kernel kernel, string serviceId)
+    /// <summary>
+    /// Invoke the prompt function to run for a specific service id.
```

Comment:
How about this description:
```
Shows how to invoke a prompt and specify the service id of the preferred AI service.
When the prompt is executed the AI Service with the matching service id will be selected.
```

Reply:
Can you also add an example showing what happens if the preferred service is not available (for both the service id and model id cases)

Reply:
Not sure about this one, as we don't use samples for showing wrong paths. I can add an assert and create a new sample for an undesired outcome though.